### PR TITLE
Added support for disabling the motor output protocol.

### PR DESCRIPTION
**Pull request by mikeller**
To go with betaflight/betaflight#9619.

image

image

### CODE REVIEW

**Comment by Asizon** [comment]
Wow really good security improvement, tested in both parts configurator and firmware and looks so good. Maybe can remove this blue text in this  Configuration part?

Escprotocol

**Comment by mikeller** [comment]
@Asizon: Thanks for reviewing, and well spotted - I split the on-connect message the wrong way.
Fixed.

File: src/js/tabs/configuration.js [hunk 1]
```diff
@@ -516,47 +521,68 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
...
 
             //trigger change unsyncedPWMSwitch to show/hide Motor PWM freq input
             $("input[id='unsyncedPWMSwitch']").change();
+        }
 
+        esc_protocol_e.val(PID_ADVANCED_CONFIG.fast_pwm_protocol + 1);
+        esc_protocol_e.change(function () {
+            const escProtocolValue = parseInt($(this).val()) - 1;
+
+            let newValue = undefined;
```
Comment (McGiverGim) [inline src/js/tabs/configuration.js]: Is this assignation needed? By default any new variable contains undefined, and the recommendation if you want to explicit assign a no value to a variable is to assign null.
Does the analytics require undefined and not null?

Comment (mikeller) [reply src/js/tabs/configuration.js]: It's not strictly needed, but in cases like these where `unassigned` is explicitly used to unassign a prior assignment (in `self.analyticsChanges['EscProtocol'] = newValue`) I find making it explicit makes it more obvious that this is intended behaviour.

**Comment by mikeller** [comment]
@McGiverGim:

> I prefer to use null and not undefined in this case. According to some articles in the web:

The problem is that a `null` value will be sent to analytics, as some metrics can legitimately report on a value being `null` - what we want here is to **not** report a value if it has not been changed, which is done by setting the metric to `undefined`.

File: src/js/tabs/configuration.js [hunk 2]
```diff
@@ -516,47 +521,68 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
...
-            $('div.checkboxPwm').toggle(!digitalProtocol);
-            $('div.unsyncedpwmfreq').toggle(!digitalProtocol);
+            $('div.digitalIdlePercent').toggle(protocolConfigured && digitalProtocol);
+            $('.escSensor').toggle(protocolConfigured && digitalProtocol);
 
-            $('div.digitalIdlePercent').toggle(digitalProtocol);
-            $('.escSensor').toggle(digitalProtocol);
+            $('div.checkboxDshotBidir').toggle(protocolConfigured && semver.gte(CONFIG.apiVersion, "1.42.0") && digitalProtocol);
+            $('div.motorPoles').toggle(protocolConfigured && semver.gte(CONFIG.apiVersion, "1.42.0"));
+            $('div.motorPoles').toggle(protocolConfigured && rpmFeaturesVisible);
```
Comment (Asizon) [inline src/js/tabs/configuration.js]: @mikeller We have problem hidding it with non digital protocol, I forgot it when created this motorPoles hidde code, can you change to this check?

```
            $('div.motorPoles').toggle(protocolConfigured && semver.gte(CONFIG.apiVersion, "1.42.0"));
            $('div.motorPoles').toggle(digitalProtocol && rpmFeaturesVisible);
```

Comment (Asizon) [reply src/js/tabs/configuration.js]: maybe we can move `protocolConfigured` and `digitalProtocol` into `rpmFeaturesVisible` and leave a single toggle line for msp version and `rpmFeaturesVisible` ??

**Comment by Asizon** [comment]
@mikeller thanks for fix it :)

**Pull request merged**